src/store/root.ts

```diff
@@ -71,3 +75,38 @@ export const useIncentiveDataSubscription = createSingletonSubscriber(() => {
 export const useGovernanceDataSubscription = createSingletonSubscriber(() => {
   return useRootStore.getState().refreshGovernanceData();
 }, 60000);
+
+let latest: V3FaucetService;
+useRootStore.subscribe(
+  (state) => state.currentMarketData,
+  async (selected) => {
+    const { setIsFaucetPermissioned: setFaucetPermissioned, jsonRpcProvider } =
```

Comment:
seems a bit weird, shouldn't there be a check if this currentMarket even is a test pool?

Reply:
It checks if testnet mode is enabled and only does the logic if it is.

The singleton subscribers are designed to run on any market/account change, and on a set interval. This logic only runs once when the market is changed and only if testnet mode is enabled and it's a v3 market.

Reply:
Saw the message to late:
> The singleton subscribers are designed to run on any market/account change, and on a set interval.

agreed

> This logic only runs once when the market is changed and only if testnet mode is enabled and it's a v3 market.

Which means it runs, no matter if you are on faucet page or not. Seems not so good and pretty inconsistent with any other page specific info no? E.g. like here: https://github.com/aave/interface/blob/main/pages/staking.staking.tsx#L32